### PR TITLE
[LITE] Reduce all op support

### PR DESCRIPTION
**Pull request by siju-samuel**

### CODE REVIEW

**Comment by siju-samuel** [comment]
@talumbau Could you please have a look at this PR. TIA

**Comment by gbaned** [comment]
@talumbau Can you please review this PR? Thanks!

**Comment by gbaned** [comment]
@siju-samuel Can you please resolve conflicts? Thanks!

**Comment by gbaned** [comment]
@siju-samuel gentle ping to resolve conflicts? Thanks!

File: tensorflow/lite/kernels/reduce_test.cc [hunk 1]
```diff
@@ -1221,5 +1250,89 @@ TEST(DynamicAnyOpTest, Scalar) {
   EXPECT_THAT(m.GetOutput<bool>(), ElementsAreArray({false}));
 }
 
+// Tests for reduce_all
+TEST(ConstAllOpTestFalse, NotKeepDims) {
```
Comment (talumbau) [inline tensorflow/lite/kernels/reduce_test.cc]: This test is fairly duplicative of the next test. Can you rename to ConstAllOpsTestMixed and use this data:

```
std::vector<bool> data = {true, true, false, false, true, true,
                            true, true,  false, false, true, true};
```

with an axis of `[0,2]`. The result will be `true, false, true`. (the extraneous `-3, -3` axis dimensions are specified in the next test so we are still testing that functionality.

Comment (talumbau) [reply tensorflow/lite/kernels/reduce_test.cc]: Also please add another version of the test described in the above comment with `keep_dims=true`. Thanks! Nice PR!

**Review by talumbau — CHANGES_REQUESTED** [review]
Overall looks good. Just needs some additional tests

**Comment by siju-samuel** [comment]
@talumbau I have updated the testcase based on your review comment. please help to review again.

File: tensorflow/lite/schema/schema_generated.h [hunk 2]
```diff
@@ -699,7 +700,8 @@ inline const BuiltinOperator (&EnumValuesBuiltinOperator())[117] {
     BuiltinOperator_QUANTIZE,
     BuiltinOperator_MATRIX_SET_DIAG,
     BuiltinOperator_ROUND,
-    BuiltinOperator_HARD_SWISH
+    BuiltinOperator_HARD_SWISH,
+    BuiltinOperator_REDUCE_ANY
```
Comment (talumbau) [inline tensorflow/lite/schema/schema_generated.h]: Should be `BuiltinOperator_REDUCE_ALL` here.

**Review by talumbau — CHANGES_REQUESTED** [review]
Thanks for adding the tests! This looks good except for the one typo I noted. I can approve as soon as that is resolved. Thanks!

**Comment by siju-samuel** [comment]
@talumbau Thanks a lot for review and pointing out the bug. I have updated the code.

**Review by talumbau — APPROVED** [review]

**Comment by siju-samuel** [comment]
@gbaned Could u please help to merge this PR before new conflicts comes. TIA

**Comment by jdduke** [comment]
Hi @siju-samuel, can you link to the models which require this operator? Thanks. The bar for adding new builtin ops has been raised, as we want to make the builtin op set a clear reference for the most important operators required for mobile execution.

**Comment by gbaned** [comment]
@siju-samuel Did you get a chance to look on jdduke's  comments? Please let us know on the update. Thanks!

**Comment by siju-samuel** [comment]
For inference purpose, currently I'm not using any model which have reduce-all op.

**Comment by gbaned** [comment]
@siju-samuel Could you please resolve the conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**